stable_horde/horde.py

```diff
@@ -238,14 +284,17 @@ async def handle_request(self, job: HordeJob):
         if job.karras:
             sampler_name += "_ka"
 
+        # Map model name to model
+        local_model = self.current_models.get(job.model, shared.sd_model)
+
         sampler = sd_samplers.samplers_map.get(sampler_name, None)
         if sampler is None:
             raise Exception(f"ERROR: Unknown sampler {sampler_name}")
 
         postprocessors = job.postprocessors
 
         params = {
-            "sd_model": shared.sd_model,
+            "sd_model": local_model,
```

Comment:
I didn't have a chance to test this locally. But I am curious how does this line work. Also are there memory or VRAM requirements when the webui switching the model?

Reply:
The local model is set a few lines earlier where I map the dictionary of available models to the requested model of the request.
And there are no vram requirements, since stable diffusion loads the models based on the set cache size in the webui settings. So for instance if I allow only 1 model at a time, I would have to load the model from the hard drive each swap. If I allow more models on cache, the swap will be quicker.